### PR TITLE
Adding UMAP analysis

### PR DESCRIPTION
**Pull request by gwaybio**
Adding all associated files to perform and visualize UMAP

You'll find this PR related to #9

### CODE REVIEW

File: 4.analyze-data/notebooks/UMAP/nbconverted/0.UMAP.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+from pycytominer import feature_select
+from pycytominer.cyto_utils import infer_cp_features
+
+
+# In[2]:
+
+
+# Set constants
+umap_random_seed = 1234
```
Comment (jenna-tomkinson) [inline 4.analyze-data/notebooks/UMAP/nbconverted/0.UMAP.py:22]: Why set the random seed to specifically 1234? I normally have seen this as 0, so I am curious the motivation.

File: 4.analyze-data/notebooks/UMAP/nbconverted/0.UMAP.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        n_components=umap_n_components
+    )
+    
+    # Remove NA columns
+    cp_df = cp_dfs[plate_name]
+    cp_df = feature_select(
+        cp_df,
+        operation="drop_na_columns",
+        na_cutoff=0
+    )
```
Comment (jenna-tomkinson) [inline 4.analyze-data/notebooks/UMAP/nbconverted/0.UMAP.py:70]: I am curious about the NA columns. Are these NA columns for features? I have not seen this before, only in the metadata when the regular expression did not work.

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! A few questions within the .py file for you.

I did want to mention, there seems to be two figures that look the same, e.g. `UMAP_localhost220512140003_KK22-05-198_treatment-checkpoint.png` and `UMAP_localhost220512140003_KK22-05-198_treatment.png`. Is there a reason to have them both?

File: 4.analyze-data/notebooks/UMAP/nbconverted/0.UMAP.py (hunk 1 above)
Comment (gwaybio) [reply 4.analyze-data/notebooks/UMAP/nbconverted/0.UMAP.py:22]: it is a random number :D

**Comment by gwaybio** [comment]
> I did want to mention, there seems to be two figures that look the same, e.g. UMAP_localhost220512140003_KK22-05-198_treatment-checkpoint.png and UMAP_localhost220512140003_KK22-05-198_treatment.png. Is there a reason to have them both?

Great find! This is my mistake. Removed the checkpoint file in 7acb3dd17b64b5fc1a2e479772f093cb9f922eed

Thanks! Will go ahead and merge now

**Pull request merged**